app/code/community/Studioforty9/Recaptcha/Block/Explicit.php

```diff
@@ -181,4 +189,47 @@ protected function _getHelper()
     {
         return Mage::helper('studioforty9_recaptcha');
     }
+
+    protected function _toHtml()
+    {
+        if (!$this->_getHelper()->isEnabled()) {
+            return '';
+        }
+
+        return parent::_toHtml();
+    }
+
+
+    private function getRecaptchaScriptV2($id): string
+    {
+        $language = Mage::app()->getLocale()->getLocale()->toString();
+        $lang = 'en';
+
+        foreach ($this->_languages as $options => $_lang) {
+            if (stristr($options, $language)) {
+                $lang = $_lang;
+            }
+        }
+
+        $query = array(
+            'onload' => 'onloadCallback' . $id,
+            'render' => 'explicit',
+            'hl' => $lang
+        );
+
+        return sprintf(
+            '<script src="https://www.google.com/recaptcha/api.js?%s" async defer></script>',
+            http_build_query($query)
+        );
+    }
+
+    private function getRecaptchaScriptV3(): string
+    {
+        $siteKey = $this->getSiteKey();
+        $id = $this->getRecaptchaId();
+
+        return <<<HTML
+<script async src="https://www.google.com/recaptcha/api.js?render=$siteKey&onload=recaptchaOnloadCallback_$id"></script>
```

Comment:
```suggestion
<script async src="https://www.google.com/recaptcha/api.js?render={$siteKey}&onload=recaptchaOnloadCallback_{$id}"></script>
```
Userei le parentesi graffe per evidenziare meglio le variabili embeddate

Reply:
Mmmm ma non si mettono così?

 ```suggestion
<script async src="https://www.google.com/recaptcha/api.js?render=${siteKey}&onload=recaptchaOnloadCallback_${id}"></script>
```

Reply:
Nell'esempio 8 [qui](https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc) non sembra. In pratica la sintassi heredoc funziona come i doppi apici quanto ad inlining delle variabili.

Reply:
perché per coerenza non utilizzi la stessa funzione come fatto per la funzione getRecaptchaScriptV2?
Tipo così:

```
$query = array(
    'render' => $siteKey,
    'onload' => 'recaptchaOnloadCallback_'.$id
);

echo sprintf(
    '<script async src="https://www.google.com/recaptcha/api.js?%s"></script>',
    http_build_query($query)
);
```

Reply:
Ah è uguale, non lo sapevo:
```
➜ psysh
Psy Shell v0.11.1 (PHP 7.4.28 — cli) by Justin Hileman
New version is available (current: v0.11.1, latest: v0.11.2)
>>> $a = 'world'
=> "world"
>>> $b = "hello ${a}"
=> "hello world"
>>> $b = "hello {$a}"
=> "hello world"
```

Reply:
Alla fine per questa cosa ho fatto [questa commit](https://github.com/webgriffe/Recaptcha/pull/1/commits/91d070f5b8360e628a6d316f6ae52212463c4133). Il metodo `getRecaptchaScript()` in v3 non veniva realmente usato. Veniva chiamato nel template ma in una variabile JS commentata. Ho quindi deciso di non usarlo del tutto in v3 e di fare un metodo che restituisce solo l'URL del JS. `getRecaptchaScript()` rimane solo per la v2 e non cambia nome per ragioni di BC.